beacon/pow/src/test/java/tech/pegasys/teku/beacon/pow/FallbackAwareEth1ProviderSelectorTest.java

```diff
@@ -114,6 +114,31 @@ void shouldFallbackOnGetLogs_smallerRangeException_timeout() {
         .isCompleted();
   }
 
+  @Test
+  void shouldContainSuppressedSocketTimeoutException() {
+    // One node failing with socket timeout error
+    when(node2.isValid()).thenReturn(false);
+    when(node3.isValid()).thenReturn(false);
+    when(node1.ethGetLogs(ethLogFilter))
+        .thenReturn(
+            failingProviderGetLogsWithError(new SocketTimeoutException("socket timeout error")));
+
+    assertThat(
+            fallbackAwareEth1Provider
+                .ethGetLogs(ethLogFilter)
+                .thenRun(() -> fail("should fail!"))
+                .exceptionallyCompose(
+                    err -> {
+                      final Throwable errCause = err.getCause();
+                      assertThat(errCause).isInstanceOf(Eth1RequestException.class);
+                      assertThat(errCause.getSuppressed().length).isEqualTo(1);
+                      assertThat(errCause.getSuppressed()[0])
+                          .isInstanceOf(SocketTimeoutException.class);
```

Comment:
given that we're doing the raising, I'd probably suggest just using
```
assertThat(err.getCause()).isInstanceOf(Eth1RequestException.class);
```

Reply:
Could potentially simplify it by getting a SafeFuture, then afterwards evaluating the failed future...
```
SafeFuture<?> future =  fallbackAwareEth1Provider.ethGetLogs(ethLogFilter);
    assertThat(future).isCompletedExceptionally();
    assertThatThrownBy(future::get)
        .hasCauseInstanceOf(Eth1RequestException.class);
```
That shows the failure like the existing test, but what it still doesn't really show is the log not containing the exception, or failing when it does contain the trace...

Reply:
Yep, the test doesn't actually check that the logs do not contain the stack trace. Was wondering if it is worth it to intercept the logger ... I will mock the logger and add a check for that

Reply:
We probably don't want to mock the logger in this case. Adding a custom appender for the test is probably the best option. https://github.com/Hakky54/log-captor may be an option to make that easy. I haven't used it before so it may not work out, but if it does work it would be quite useful as it seems to provide a pretty simple approach to testing log output which we could use in other places.

Reply:
It works and intercepts almost everything! Added it as a dependency for the beacon module only for now. We could add it elsewhere if required